### PR TITLE
Automatically fetch dependencies in pom

### PR DESCRIPTION
**Pull request by THenry14**
## Linked issues

Closes https://github.com/software-mansion/starknet-jvm/issues/143

## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

File: lib/build.gradle.kts [hunk 1]
```diff
@@ -192,27 +192,16 @@ publishing {
...
-                    okHttpDependency.appendNode("artifactId", "okhttp")
-                    okHttpDependency.appendNode("version", "4.10.0")
-
-                    val kotlinxDependency = dependencyNode.appendNode("dependency")
-                    kotlinxDependency.appendNode("groupId", "org.jetbrains.kotlinx")
-                    kotlinxDependency.appendNode("artifactId", "kotlinx-serialization-json")
-                    kotlinxDependency.appendNode("version", "1.3.3")
+                    val dependenciesNode = asNode().appendNode("dependencies")
+
+                    configurations["implementation"].allDependencies.forEach {
```
Comment (cptartur) [inline lib/build.gradle.kts:197]: I assume this collects everything that was declare as `implementation ...` in the build.gradle.kts?

**Review by cptartur — COMMENTED** [review]
How was this tested? I assume there's not simple automatic way to do so.

File: lib/build.gradle.kts (hunk 1 above)
Comment (THenry14) [reply lib/build.gradle.kts:197]: yes

**Comment by THenry14** [comment]
> How was this tested? I assume there's not simple automatic way to do so.

`./gradlew generatePomFileForStarknetPublication`

**Review by cptartur — APPROVED** [review]
Alright, can't spot any immediate problems with this one so I guess it's okay 👍

**Pull request merged**